R/process_sc_all_care_home.R

```diff
@@ -237,38 +241,61 @@ process_sc_all_care_home <- function(
 
   # Continuous Care Home Stays
   # Stay will be continuous as long as the admission date is the next day or
-  # earlier than the previous discharge date
-
+  # earlier than the previous discharge date.
+  # creates a CIS  flag for CHI across all of scotland
+  # and a CIS for social care ID and sending location for just that LA
   ch_markers <- matched_deaths_data %>%
-    # ch_chi_cis
+    # uses the chi to flag continuous stays. Will flag cases even if in another LA
     dplyr::group_by(.data[["chi"]]) %>%
-    dplyr::mutate(
-      continuous_stay_chi = tidyr::replace_na(
-        .data[["ch_admission_date"]] <= dplyr::lag(
-          .data[["ch_discharge_date"]]
-        ) + lubridate::days(1L),
-        TRUE
-      ),
-      ch_chi_cis = cumsum(.data[["continuous_stay_chi"]])
+    # create variable for previous discharge date + 1 day
+    dplyr::mutate(previous_discharge_date_chi = dplyr::lag(.data[["ch_discharge_date"]]) + lubridate::days(1L)) %>%
+    # TRUE/FALSE flag for if admission date is before or equal to previous discharge date + 1 day
+    dplyr::mutate(continuous_stay_flag_chi = tidyr::replace_na(.data[["ch_admission_date"]] <= previous_discharge_date_chi, FALSE)) %>%
+    # different to code in above sections.
+    # we want to uniquely identify all cases where the flag is FALSE. and only the first case where the flag is TRUE
+    # to do this create a variable of the flag in the previous row
+    dplyr::mutate(previous_continuous_stay_flag_chi = tidyr::replace_na(dplyr::lag(.data[["continuous_stay_flag_chi"]]), FALSE)) %>%
+    dplyr::mutate(continuous_stay_chi = ifelse(continuous_stay_flag_chi == FALSE |
+      (continuous_stay_flag_chi == TRUE & previous_continuous_stay_flag_chi == FALSE), FALSE, TRUE)) %>%
+    dplyr::group_by(
+      .data[["chi"]],
+      .data[["continuous_stay_chi"]]
     ) %>%
-    dplyr::ungroup() %>%
-    # ch_sc_id_cis
-    # uses the social care id and sending location so can be used for
+    # gives cases their unique CIS identifier
+    dplyr::mutate(ch_chi_cis = ifelse(continuous_stay_chi == FALSE, dplyr::row_number(), NA)) %>%
+    dplyr::group_by(
+      .data[["social_care_id"]],
+      .data[["sending_location"]]
+    ) %>%
+    # fills in CIS identifier for all cases
+    tidyr::fill(ch_chi_cis, .direction = c("down")) %>%
+    # This is the same but uses the social care id and sending location so can be used for
     # episodes that are not attached to a CHI number
     # This will restrict continuous stays to each Local Authority
     dplyr::group_by(.data[["social_care_id"]], .data[["sending_location"]]) %>%
```

Comment:
```suggestion
    sc_ch_id_markers <- chi_chi_markers %>%
    dplyr::group_by(.data[["social_care_id"]], .data[["sending_location"]]) %>%
```
I think the sc_id_cis can be split onto a new object just to make it a bit clearer that its separate even though it uses the same logic as above. 

Alternatively we could put the logic into a function since we are repeating this for sc_id_cis